data-prepper-core/src/main/java/org/opensearch/dataprepper/peerforwarder/server/PeerForwarderHttpServerProvider.java

```diff
@@ -61,10 +62,16 @@ public Server get() {
             );
 
             if (peerForwarderConfiguration.getAuthentication() == ForwardingAuthentication.MUTUAL_TLS) {
-                sb.tlsCustomizer(sslContextBuilder -> sslContextBuilder.trustManager(
-                                new ByteArrayInputStream(certificate.getCertificate().getBytes(StandardCharsets.UTF_8))
-                        )
-                        .clientAuth(ClientAuth.REQUIRE));
+                if (peerForwarderConfiguration.isSslFingerprintVerificationOnly()) {
```

Comment:
How is fingerPrintVerification option set ? 

Unit test for this change ? Not a blocker, we can follow up on this.

Reply:
It's part of the peer_forwarder config in the data-prepper-config.yaml file:
https://github.com/opensearch-project/data-prepper/blob/main/docs/peer_forwarder.md.

I went to add unit tests but the existing class for it is pretty bare bones: https://github.com/opensearch-project/data-prepper/blob/main/data-prepper-core/src/test/java/org/opensearch/dataprepper/peerforwarder/server/PeerForwarderHttpServerProviderTest.java

This code is covered by the IT in this class though: https://github.com/opensearch-project/data-prepper/blob/main/data-prepper-core/src/test/java/org/opensearch/dataprepper/peerforwarder/PeerForwarder_ClientServerIT.java#L404-L444